File: lp2/PokemonV-C#/PokemonV/Potion.cs
using System;

namespace PokemonV
{
    public class Potion : Item
    {
        public Potion(string name, string descrip, int points)
            : base(name, descrip)
        {
            this.healthPoints = points;
        }

        public void pour()
        {

        }

        protected int healthPoints;
        public int HealthPoints
        {
            get { return healthPoints; }
            set { healthPoints = value; }
        }
    }
}

File: lp2/PokemonV-Client/src/com/grupox/pokemonv/model/Potion.java
package com.grupox.pokemonv.model;

public class Potion extends Item{
    /* Attributes */
    private int hp;

    /* Constructors */
    public Potion(int hp, int quantity){
        super(quantity);
        this.hp = hp;
        this.name = "Potion";
    }
    
    /* Getters && Setters */
    public int getHp() {
        return hp;
    }
    public void setHp(int hp) {
        this.hp = hp;
    }
}